eh_win32_msvc.cc

```diff
@@ -105,6 +112,9 @@ OBJC_PUBLIC extern "C" void objc_exception_rethrow(void* exc);
 
 OBJC_PUBLIC extern "C" void objc_exception_throw(id object)
 {
+	// Set unhandled exception filter to support _objc_unexpected_exception hook
+	SetUnhandledExceptionFilter(&_objc_unhandled_exception_filter);
```

Comment:
Do we need to do this every time an exception is thrown, or can we do it once on init time?  If we do need to do it here, can you add a comment explaining why please?

Reply:
As I mentioned in the issue description, I tried installing the handler in a constructor block (using `__attribute__((constructor))`), but this seems to get called before the VC runtime gets initialized, which installs its own hook and thus ours gets overwritten.

I’m not very happy calling this on every exception, as it means that throwing an exception has the side effect of overwriting any existing exception handler, and as far as I can see there’s no way for us to restore it as we can’t execute code after an exception is possibly handled. However I couldn’t come up with any other solution so this is what I went with.

For now I added a comment to explain the above, but if you can think of any other way to do this please let me know.

Reply:
Since `SetUnhandledExceptionFilter()` returns the previous handler (if any), I now save that and call it as part of our handler. Does that sound ok to you?

The only issue I see with this is that if another piece of code would later try to unset the handler it had previously set (and which we stored) that would no longer work, but since handlers can get overwritten it seems like it’s something that any code using these handler should have to account for anyway.

Reply:
That sounds sensible to me.